buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy

```diff
@@ -287,6 +341,20 @@ class VagrantTestPlugin implements Plugin<Project> {
         createPlatformTestTask(project)
     }
 
+    private static Map<String, String> vagrantEnvVars(Project project) {
+        /*
+         * We always use the main project.rootDir as Vagrant's current working directory (VAGRANT_CWD)
+         * so that boxes are not duplicated for every Gradle project that use this VagrantTestPlugin.
+         */
+        def vagrantEnvVars = [
+                'VAGRANT_CWD'        : "${project.rootDir.absolutePath}",
+                'VAGRANT_VAGRANTFILE': 'Vagrantfile',
+                'VAGRANT_PROJECT_DIR': "${project.projectDir.absolutePath}"
+        ]
+        vagrantEnvVars.putAll(VAGRANT_ENV_VARS)
```

Comment:
I think it is weird to have these named the same way. Maybe look in `WINDOWS_BOXES` instead? Or name `VAGRANT_ENV_VARS` to `VAGRANT_BOX_ENV_VARS` or something.

Reply:
I agree, I'll make the naming more clear.

Actually originally I wrote it like

```
def vagrantEnvVars = [ ... ]
if ('windows-2016' in WINDOWS_BOXES) {
  vagrantEnvVars.put('VAGRANT_WINDOWS_2016_BOX', WINDOWS_BOXES['windows-2016'])
}
// same for windows 2012r2
return vagrantEnvVars
```

but changed it to just store the env vars directly because it seemed like an unnecessary level of indirection. What do you think?

Reply:
I'm fine with digging through the `WINDOWS_BOXES` though I might want  add something that fails if there is an unknown key in there. It'd be inconvenient to just add something to that Map and forget to update this method and have the box fail to show up mysteriously.